main.go

```diff
@@ -87,14 +90,23 @@ func getBlackholeWriter(tr *throughputRecorder) func(a ...any) (int, error) {
 
 type OutFunc func(a ...any) (int, error)
 
+// rustWasm was compiled using `cargo build --release --target wasm32-wasi`
+// VRL currently cannot build on wasm32-unknown-unknown, so we target wasm32-wasi
```

Comment:
This is interesting. We actually have VRL running as `wasm32-unknown-unknown` (see https://playground.vrl.dev). Would you mind elaborating what stopped you from building it?

It is true that some functions need OS-level access, which requires `wasm32-wasi`, but unless you need those specific functions, you should be good.

See also: https://github.com/vectordotdev/vector/tree/d620352924ac1253c9f0fcace49584173a0dcd30/lib/vrl/web-playground

Reply:
Looks like the `getrandom` crate dependency is what fails
```
error: the wasm32-unknown-unknown target is not supported by default, you may need to enable the "js" feature. For more information see: https://docs.rs/getrandom/#webassembly-support
   --> /home/lima.linux/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.2.8/src/lib.rs:263:9
    |
263 | /         compile_error!("the wasm32-unknown-unknown target is not supported by \
264 | |                         default, you may need to enable the \"js\" feature. \
265 | |                         For more information see: \
266 | |                         https://docs.rs/getrandom/#webassembly-support");
    | |________________________________________________________________________^

error[E0433]: failed to resolve: use of undeclared crate or module `imp`
   --> /home/lima.linux/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.2.8/src/lib.rs:290:5
    |
290 |     imp::getrandom_inner(dest)
    |     ^^^ use of undeclared crate or module `imp`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `getrandom` due to 2 previous errors
```

Their docs say you can target `wasm32-unknown-unknown` by specifying the `js` feature which will use a Javascript API to get randomness, which doesn't really solve the problem here. (Unless the `getrandom` crate isn't used in the codepaths we need, I haven't gotten this far in my testing)

Reply:
There's a list of functions prevent VRL from compiling with [`wasm32-unknown-unknown` tracked here]( https://github.com/vectordotdev/vector/issues?q=wasm+compatible+label%3A%22vrl%3A+playground%22+is%3Aopen)

Adding the `js` feature in the cargo.toml will allow it to compile for some of these functions but there are still others, such as `reverse_dns()`, that fail to compile for reasons other than `getrandom()`, probably due to the sys module not existing in `wasm32-unknown-unknown`.

I believe for the purpose of running this with the agent it's perfectly fine to use `wasm32-wasi` since `wasm32-unknown-unknown` is meant to be restrictive so it can run in the web. 

In the future I see myself making enough progress throughout the playground where I'll introduce conditional compilation based on targets so that VRL stdlib compiles [without specifying the functions to use in cargo.toml like we had to do for the playground here](https://github.com/vectordotdev/vector/blob/f94aaed201f373b03ced3273cbd24d2a4d33c8da/lib/vrl/web-playground/Cargo.toml#L20)

Reply:
To add to @jonathanpv's comment. You can compile VRL without the default functions enabled, and then enable them individually, to remove the ones that don't compile on wasm32-unknown-unknown.

Here's an example on how to do so:

https://github.com/vectordotdev/vector/blob/d620352924ac1253c9f0fcace49584173a0dcd30/lib/vrl/web-playground/Cargo.toml#L20-L153

Reply:
I believe I've already configured that, unless I made a mistake in my implementation
https://github.com/gh123man/cgo-rust-perf-test/pull/3/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R35-R42

I was surprised to see that `getrandom` is still pulled in even with minimal VRL features pulled in

Reply:
> I believe I've already configured that, unless I made a mistake in my implementation https://github.com/gh123man/cgo-rust-perf-test/pull/3/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R35-R42
> 
> I was surprised to see that `getrandom` is still pulled in even with minimal VRL features pulled in

Ah you're right, in this case you found another function I need to add to the issues haha. Weird, could've sworn it compiled before and didn't pull in getrandom. 

I also find myself having to hunt down which crates are called, perhaps a crate dependency crawling tool should be made to see if it will compile to  specific targets we want.